File: AvitoTest/app/src/main/java/avito/test/data/MyEndPoint.java
package avito.test.data;

import retrofit.Endpoint;

public class MyEndPoint implements Endpoint {

  private String url;

  public MyEndPoint(String serverUrl) {
    url = serverUrl;
  }

  public void setUrl(String url) {
    this.url = url;
  }

  @Override public String getUrl() {
    if (url == null) throw new IllegalStateException("url not set.");
    return url;
  }

  @Override public String getName() {
    return "default";
  }
}

File: AvitoTest/app/src/main/java/avito/test/data/MyEndPoint.kt
package avito.test.data

import retrofit.Endpoint

public class MyEndPoint(private var url: String?) : Endpoint {

    public fun setUrl(url: String) {
        this.url = url
    }

    override fun getUrl(): String {
        if (url == null) throw IllegalStateException("url not set.")
        return url
    }

    override fun getName(): String {
        return "default"
    }
}